preflight-checks/src/checks/02-check-port-and-healthcheck-route.js

```diff
@@ -0,0 +1,65 @@
+import chalk from 'chalk';
+import packageJson from '../package';
+const execa = require( 'execa' );
+const waait = require( 'waait' );
+const fetch = require( 'isomorphic-fetch' );
+
+const CACHE_HEALTHCHECK_ROUTE = '/cache-healthcheck?';
+const chalkNpmStart = chalk.yellow( 'npm start' );
+const chalkPORT = chalk.yellow( 'PORT' );
+const chalkHealthCheckRoute = chalk.yellow( CACHE_HEALTHCHECK_ROUTE );
+const chalkVIPGo = chalk.yellow( '@automattic/vip-go' );
+
+module.exports = {
+	name: `Building the app and running ${ chalkNpmStart }...`,
+	excerpt: `Checking if your app accepts a ${ chalkPORT } and responds to ${ chalkHealthCheckRoute }`,
+	run: async () => {
+		const scripts = packageJson.scripts;
+		const PORT = Math.floor( Math.random() * ( 4000 - 3000 ) + 3000 ); // Get a PORT between 3000 and 4000
+
+		let subprocess;
+
+		console.log( chalk.blue( '  Info:' ), `Installing dependencies with ${ chalk.yellow( 'npm install' ) }...` );
+
+		return execa.shell( 'npm install' )
```

Comment:
yeah, that's better. I used `rmfr` which is a wrapper around `rimraf` to make it promise based

Reply:
I also fixed the `--production` flag

Reply:
looking closer into this, we actually need both. To build the app we need devDependencies, and to run it (and mimic VIP Go) we need production dependencies. So I guess the proper workflow is:
- `npm install`
- `npm build`
- `rm -fr node_modules`
- `npm install --production`
- `npm start`

Reply:
I think It should probably be this:

```
rm -fr node_modules

if build-ci script exists {
   npm install
   npm run build-ci
   rm -fr node_modules
}

npm install --production
npm run build
npm start
```

Reply:
For the build command `npm run build` (and in case there is no `build-ci`), we actually need `devDependencies` to build the app, so we can't pass the `--production` flag to `npm install`